drivers/hmis/app/graphql/mutations/ac_hmis/create_outgoing_referral_posting.rb

```diff
@@ -0,0 +1,70 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class AcHmis::CreateOutgoingReferralPosting < CleanBaseMutation
+    description 'Create outgoing referral posting'
+
+    argument :input, Types::HmisSchema::OutgoingReferralPostingInput, required: false
+
+    field :record, Types::HmisSchema::ReferralPosting, null: true
+    field :errors, [Types::HmisSchema::ValidationError], null: false, resolver: Resolvers::ValidationErrors
+
+    def resolve(input:)
+      handle_error('connection not configured') unless HmisExternalApis::AcHmis::LinkApi.enabled?
+
+      enrollment = Hmis::Hud::Enrollment
+        .viewable_by(current_user)
+        .find_by(id: input.enrollment_id)
+      handle_error('enrollment not found') unless enrollment
+      handle_error('access denied') unless current_user.can_manage_outgoing_referrals_for?(enrollment.project)
+
+      project = Hmis::Hud::Project
+        .viewable_by(current_user)
+        .find_by(id: input.project_id)
+      handle_error('project not found') unless project
+
+      referral = HmisExternalApis::AcHmis::Referral.new(
+        enrollment: enrollment,
+        referral_date: Time.current,
+        service_coordinator: current_user.name,
+      )
+      referral.household_members = enrollment.household_members.preload(:client).map do |member|
+        HmisExternalApis::AcHmis::ReferralHouseholdMember.new(
+          relationship_to_hoh: member.relationship_to_hoh,
+          client_id: member.client.id
+        )
+      end
+
+      posting = referral.postings.build(
+        status: 'assigned_status',
+        project: project,
+        unit_type_id: input.unit_type_id,
```

Comment:
I think we should error if the unit type is not available in the project. Something like: `project.units.unoccupied_on.pluck(:unit_type_id).uniq`

Reply:
added validation per our discussion